### PR TITLE
Add mjml-core package with initial typings

### PR DESCRIPTION
**Pull request by IanEdington**
- Adds more details about the options for `mjml2html`
- Adds the types for MJML custom components
- rearrange types to more accurately describe interaction between `mjml` and `mjml-core`

## ✅ mjml re-export mjml-core or mjml-core re-export mjml
Based on the implementation, mjml re-exports `mjml2html` from mjml-core. However, most users only use mjml. This change will require both @types/mjml and @types/mjml-core to be added when most users just need @types/mjml.

I'm looking for an opinion from a maintainer on the best way to handle this. The technically more correct way or the way that's better for the end user.

Decision: have mjml re-export mjml-core the way the JS behaves
Based on conversation in discord

## Excluded functions
These three functions don't seem to be useful outside the MJML packages and so weren't included here. I'm happy to include them if they should be included.
- initComponent
- initializeType
- handleMjmlConfig

### CODE REVIEW

File: types/mjml-core/index.d.ts
Comment (IanEdington) [inline types/mjml-core/index.d.ts:7]: @aahoughton @marpstar @eiskalteschatten @emrah88 I've added you as maintainers on mjml-core since you contributed to mjml and it's basically the same package, with a few additions.

**Comment by danger-public** [comment]
Inspecting the JavaScript source for this package found some properties that are not in the .d.ts files.
The check for missing properties isn't always right, so take this list as advice, not a requirement.
## mjml-core ([<kbd>unpkg</kbd>](https://unpkg.com/browse/mjml-core@latest/))
was missing the following properties:
1. initComponent
1. initializeType
1. handleMjmlConfig
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 54320e7c3d8aea42802af155768ccfc9b278f19c
</p>

File: types/mjml/index.d.ts
Comment (IanEdington) [inline types/mjml/index.d.ts:12]: reverting this based on https://github.com/DefinitelyTyped/DefinitelyTyped#a-package-uses-export--but-i-prefer-to-use-default-imports-can-i-change-export--to-export-default

**Review by ryanburr — APPROVED** [review]
Looks good to me, but I'm not a maintainer so I'm not sure if my approval means anything

**Comment by IanEdington** [comment]
I heard back from a Definitely Typed maintainers in Discord. It sounds like this is the approach we should take with mjml / mjml-core types since anyone with mjml will have a hard depencency on mjml-core.
That was the last open question from my perspective. IMO this is ready to merge

@ngarnier As a maintainer of mjml, could you give this a :stamp: if these types describe the mjml and mjml-core interface? Or feedback on what you would like to see?

@TrySound you might also be a good person to review these types

File: types/mjml-core/index.d.ts
Comment (iRyusa) [inline types/mjml-core/index.d.ts:139]: I think we can type this a bit better
Either: 
- no children/no content (ex mj image)
- children but no content
- content but no children

Comment (IanEdington) [reply types/mjml-core/index.d.ts:139]: great suggestion! I'll do that

**Comment by IanEdington** [comment]
@iRyusa I'll make the changes you suggested this weekend.

@aahoughton, @marpstar, @eiskalteschatten, @emrah88 Are you still interested in being maintainers on this package? If you 👎 this comment I'll remove you from the list.

Otherwise do you have any feedback beyond what @iRyusa gave. I'd like to be able to make any required changes together this weekend.

**Comment by eiskalteschatten** [comment]
I'm no longer involved with Mailjet, so it doesn't make sense for me to continue maintaining it...

**Comment by IanEdington** [comment]
@eiskalteschatten I've removed you from both packages. Thank you for your service 🙏 

@ryanburr I've added you as a maintainer since you're been the most active on this project with me. Please let me know if you would like to be removed.

@iRyusa I believe I've made the changes you asked for. Let me know if they align with what you were thinking. Also, let me know if you would be willing to be added as a maintainer on these types.

@aahoughton, @marpstar, @emrah88 It's been a 20 days that this PR has been out and we haven't heard from you. Would you like me to remove you as maintainers?

**Comment by IanEdington** [comment]
@ryanburr @iRyusa Can you approve these changes?

File: types/mjml-core/index.d.ts
Comment (ryanburr) [inline types/mjml-core/index.d.ts:147]: Just so I'm clear, these would be examples of each JSON interface respectively,
```html
<mj-section>
  <mj-column>
    ...
  </mj-column>
</mj-section>
```

```html
<mj-text>
  My text content here
</mj-text>
```

```html
<mj-image />
```

Is this correct?

**Review by ryanburr — COMMENTED** [review]
@IanEdington Just 1 clarification

File: types/mjml-core/index.d.ts
Comment (IanEdington) [reply types/mjml-core/index.d.ts:147]: Correct! Do you have a suggestion for better names to make this more clear? Or is there a comment you think would be valuable?

Comment (ryanburr) [reply types/mjml-core/index.d.ts:147]: What do you think about the following?

`MJMLJsonWithChildren`
`MJMLJsonWithContent`
`MJMLJsonSelfClosingTag`

Comment (IanEdington) [reply types/mjml-core/index.d.ts:147]: Thanks, Great suggestion! Done

**Review by ryanburr — APPROVED** [review]

File: types/mjml-core/index.d.ts
Comment (iRyusa) [reply types/mjml-core/index.d.ts]: 👍

**Review by iRyusa — APPROVED** [review]

**Comment by orta** [comment]
OK, looks good to me and we have general approval too 👍🏻

**Pull request merged**
